### PR TITLE
* dnsforward, config: add unspecified IP blocking option

### PR DESCRIPTION
**Pull request by alexzeitgeist**
This feature adds support to answer filtered queries with the unspecified address (0.0.0.0), also referred to as [NULL blocking mode in pi-hole](https://docs.pi-hole.net/ftldns/blockingmode/). Also see [issue #742 ](https://github.com/AdguardTeam/AdGuardHome/issues/742).

At least in my non-scientific testing on an Android device, this mode works better than the default NXDOMAIN filtering, as it reduces the number of repeated dns queries.

It is enabled by adding `null_filter: true` to the dns section of AdGuardHome.yaml, for example:
```
...
dns:
  bind_host: 0.0.0.0
  port: 53
  protection_enabled: true
  filtering_enabled: true
  null_filter: true
...
```

~~Note that also host entries in filter lists with 127.0.0.1 will be answered with the unspecified address. This is intentional as 0.0.0.0 seems more compatible accross different operation systems than 127.0.0.1.~~

Lastly, an option for this feature would still have to be added to the UI frontend. I had some difficulty adding the React code to my Goland project for debugging.. :p

### CODE REVIEW

**Comment by ameshkov** [comment]
@alexzeitgeist thank you!

@szolin plz take a look

**Comment by szolin** [comment]
I think it would be better to have a string-value setting `blocking_mode` instead of boolean setting `null_filter`.

for example:

`blocking_mode: nxdomain`

and

`blocking_mode: null_ip`

This allows us to extend the functionality without having configuration issues with incompatibile settings.

File: dnsforward/dnsforward.go [hunk 1]
```diff
@@ -26,6 +26,8 @@ const (
 	parentalBlockHost     = "family-block.dns.adguard.com"
 )
 
+const unspecifiedAddress = "0.0.0.0"
```
Comment (szolin) [inline dnsforward/dnsforward.go]: Please use `net.IPv4zero` instead of this.
This will also allow us to remove `net.ParseIP()` call below and pass it directly to `genARecord()`.

Comment (alexzeitgeist) [reply dnsforward/dnsforward.go]: @szolin didn't know about `net.IPv4zero`. Fixed.

Also added a test for the null filter.

Not sure if it's best practice to hardcode the value like `null_ip`, or if it should be added as a constant.
